draft-thomson-train-protocol.md

```diff
@@ -401,14 +401,78 @@ Similarly, if there is a strong need to ensure that a rate limit is respected,
 network elements cannot assume that the signaled limit will be respected by
 endpoints.
 
-## Privacy {#privacy}
+# Privacy considerations {#privacy}
 
-Any network element that can observe the content of that packet can read the
-rate limit that was applied.
-
-{:aside}
-> TODO: Write more.
+In our privacy analysis, we studied whether TRAIN could be used by attackers
+to gain information about the ongoing traffic and compromise the privacy
+of clients or servers by exposing their identity of their activity.
 
+Any network element that can observe the content of that packet can read the
+rate limit that was applied.  Any signal is visible on the path, from the point
+at which it is applied to the point at which it is consumed at an endpoint.
+On path elements can also alter the TRAIN signal to try trigger specific
+reactions and gain further knowledge.
+
+In the general case of a client connected to a server through the
+Internet, we believe that TRAIN does not provide much advantage to attackers.
+The identities of the clients and servers are already visible through their
+IP addresses. Traffic analysis tools already provide more information than
+the data rate limits set by TRAIN. However, we are concerned with two avenues
+of attacks: that the passive observation
+of TRAIN might help identify the end points; and that the active manipulation
+of TRAIN signals might help find out the identity of clients or servers
+located behind VPNs or proxies.
+
+## Passive observation of TRAIN for fingerprinting endpoints
+
+If only few clients and server pairs negotiate the usage of TRAIN, the
+occasional observation of TRAIN packets will "stick out". That observation,
+could be combined with observation of timing and volume of traffic to
+help identify the endpoint or categorize the application that they
+are using.
+
+A variation of this issue occurs if TRAIN is widely implemented, but
+only used in some specific circumstances. In that case, observation of
+TRAIN packets reveals information about the state of the endpoint.
+
+If multiple servers are accessed through the same front facing server,
+Encrypted Client Hello (ECH) may be used to prevent outside parties to
+identify which specific server a client is using. However, if only
+a few of these servers use TRAIN, the observation of TRAIN packets
+will help identify which specific server a client is using.
+
+This issue will be mitigated if TRAIN becomes widely implemented, and
+if the usage of TRAIN is not limited to the type of applications
+that make active use of the signal.
+
+QUIC implementations are therefore encouraged to make the feature available
+unconditionally.  Endpoints might send TRAIN packets whenever a peer can accept
+them.
+
+## Active manipulation of TRAIN to identify endpoint
```

Comment:
```suggestion
## Active Attacks
```

Reply:
Github does not let me accept the suggestion. I will do it in a separate commit.